collect_app/src/test/java/org/odk/collect/android/fragments/dialogs/FirstLaunchActivityTest.kt

```diff
@@ -0,0 +1,111 @@
+package org.odk.collect.android.fragments.dialogs
+
+import android.content.Context
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.intent.Intents
+import androidx.test.espresso.intent.Intents.intended
+import androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.notNullValue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.odk.collect.android.R
+import org.odk.collect.android.activities.MainMenuActivity
+import org.odk.collect.android.application.Collect
+import org.odk.collect.android.injection.config.AppDependencyModule
+import org.odk.collect.android.preferences.source.SettingsProvider
+import org.odk.collect.android.projects.AutomaticProjectCreatorDialog
+import org.odk.collect.android.projects.ManualProjectCreatorDialog
+import org.odk.collect.android.projects.ProjectImporter
+import org.odk.collect.android.storage.StoragePathProvider
+import org.odk.collect.android.support.CollectHelpers
+import org.odk.collect.android.utilities.TranslationHandler.getString
+import org.odk.collect.android.version.VersionInformation
+import org.odk.collect.projects.ProjectsRepository
+
+@RunWith(AndroidJUnit4::class)
+class FirstLaunchActivityTest {
+    @Test
+    fun `The main menu should be started after clicking on the 'Try a demo' link`() {
+        val scenario = ActivityScenario.launch(FirstLaunchActivity::class.java)
+        scenario.onActivity {
+            Intents.init()
+            onView(withText(R.string.try_demo)).perform(click())
+            intended(hasComponent(MainMenuActivity::class.java.name))
+            Intents.release()
+        }
+    }
+
+    @Test
+    fun `Importing default project should be triggered after clicking on the 'Try a demo' link`() {
+        val projectImporter = mock(ProjectImporter::class.java)
+        CollectHelpers.overrideAppDependencyModule(object : AppDependencyModule() {
+            override fun providesProjectImporter(
+                    projectsRepository: ProjectsRepository,
+                    storagePathProvider: StoragePathProvider,
+                    context: Context,
+                    settingsProvider: SettingsProvider
+            ): ProjectImporter? {
+                return projectImporter
+            }
+        })
+
+        val scenario = ActivityScenario.launch(FirstLaunchActivity::class.java)
```

Comment:
I'd prefer to use a rule rather than explicitly building the scenario every time but I can't figure out how to do the dependency override in that case. Ok to leave like this for now?